lib/neovim-plugin.nix

```diff
@@ -29,7 +29,12 @@ with lib;
       colorscheme ? name,
       # options
       originalName ? name,
-      defaultPackage,
+      # WARNING: `defaultPackage` is being replaced with `packageOption`,
+      # which should be defined using: lib.mkPackageOption pkgs originalName { default = [ "path" "to" "package" ]; }
+      # NOTE: We can't do that here (without importing a module) because we don't have access to the `pkgs` module arg.
```

Comment:
Should this be tackled in this PR? I.e. implementing the package option within an imported-module

Reply:
It would not need to much additional complexity with #2140 right?